File: src/sugtao4423/lod/dialog_onclick/Dialog_talk.java
package sugtao4423.lod.dialog_onclick;

import twitter4j.Status;
import twitter4j.TwitterException;
import android.app.AlertDialog;
import android.content.Context;
import android.os.AsyncTask;
import android.view.View;
import android.view.View.OnClickListener;
import sugtao4423.lod.ApplicationClass;
import sugtao4423.lod.ListViewListener;
import sugtao4423.lod.R;
import sugtao4423.lod.ShowToast;
import sugtao4423.lod.tweetlistview.TweetListAdapter;
import sugtao4423.lod.tweetlistview.TweetListView;

public class Dialog_talk implements OnClickListener{

	private Status status;
	private Context context;
	private twitter4j.Status reply;

	private TweetListAdapter resultAdapter;
	private AlertDialog dialog;

	public Dialog_talk(Status status, Context context, AlertDialog dialog){
		this.status = status;
		this.context = context;
		this.dialog = dialog;
	}

	@Override
	public void onClick(View v){
		dialog.dismiss();
		TweetListView result = new TweetListView(context);
		resultAdapter = new TweetListAdapter(context);
		resultAdapter.setOnItemClickListener(new ListViewListener());
		resultAdapter.setOnItemLongClickListener(new ListViewListener());
		result.setAdapter(resultAdapter);
		new AlertDialog.Builder(context).setView(result).show();

		if(status.isRetweet())
			reply = status.getRetweetedStatus();
		else
			reply = status;
		resultAdapter.add(reply);

		new LoadConversation().execute();
	}

	private class LoadConversation extends AsyncTask<Void, Void, Status>{
		@Override
		protected twitter4j.Status doInBackground(Void... params){
			try{
				reply = ((ApplicationClass)context.getApplicationContext()).getTwitter().showStatus(reply.getInReplyToStatusId());
				return reply;
			}catch(TwitterException e){
				return null;
			}
		}

		@Override
		public void onPostExecute(twitter4j.Status result){
			if(result != null){
				resultAdapter.add(result);
				if(result.getInReplyToStatusId() > 0)
					new LoadConversation().execute();
				else
					new ShowToast(context.getApplicationContext(), R.string.success_getTalkList);
			}else{
				new ShowToast(context.getApplicationContext(), R.string.error_getTalkList);
			}
		}
	}
}

File: app/src/main/java/sugtao4423/lod/dialog_onclick/Dialog_talk.kt
package sugtao4423.lod.dialog_onclick

import android.content.Context
import android.view.View
import androidx.appcompat.app.AlertDialog
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import sugtao4423.lod.App
import sugtao4423.lod.R
import sugtao4423.lod.ShowToast
import sugtao4423.lod.tweetlistview.TweetListAdapter
import sugtao4423.lod.tweetlistview.TweetListView
import twitter4j.Status
import twitter4j.TwitterException

class Dialog_talk(status: Status, private val context: Context, private val dialog: AlertDialog) : View.OnClickListener {

    private var reply = if (status.isRetweet) status.retweetedStatus else status
    private val resultAdapter = TweetListAdapter(context)

    override fun onClick(v: View?) {
        dialog.dismiss()
        val result = TweetListView(context)
        result.adapter = resultAdapter
        AlertDialog.Builder(context).setView(result).show()

        resultAdapter.add(reply)

        loadConversation()
    }

    private fun loadConversation() {
        CoroutineScope(Dispatchers.Main).launch {
            val result = withContext(Dispatchers.IO) {
                try {
                    reply = (context.applicationContext as App).getTwitter().showStatus(reply.inReplyToStatusId)
                    reply
                } catch (e: TwitterException) {
                    null
                }
            }
            if (result != null) {
                resultAdapter.add(result)
                if (result.inReplyToStatusId > 0) {
                    loadConversation()
                } else {
                    ShowToast(context.applicationContext, R.string.success_get_talk_list)
                }
            } else {
                ShowToast(context.applicationContext, R.string.error_get_talk_list)
            }
        }
    }

}